internal/s3_folder.go

```diff
@@ -65,10 +66,7 @@ func (folder *S3Folder) ReadObject(objectRelativePath string) (io.ReadCloser, er
 }
 
 func (folder *S3Folder) GetSubFolder(subFolderRelativePath string) StorageFolder {
-	if !strings.HasSuffix(subFolderRelativePath, "/") {
-		subFolderRelativePath = subFolderRelativePath + "/"
-	}
-	return NewS3Folder(folder.uploader, folder.S3API, *folder.Bucket, folder.Path+subFolderRelativePath)
+	return NewS3Folder(folder.uploader, folder.S3API, *folder.Bucket, path.Join(folder.Path, subFolderRelativePath) + "/")
```

Comment:
Is usage of path.Join valid here? What happens here if we use some other platform, like Windows?

Reply:
I've replaced path.join with hand-written, possibly more buggy, but platfrom-independent join